test/functional/ui/screen_basic_spec.lua

```diff
@@ -77,20 +77,25 @@ describe('Screen', function()
         eq(true, screen.suspended)
       end
 
-      -- when we use eval in the test, nvim instance is already foregrounded
-      -- after being suspended, therefore both events are triggered here.
-      command('autocmd VimSuspendPre  * :let g:foo=42')
-      command('autocmd VimResumed  * :let g:bar=24')
+      -- using eval after suspending, resumes nvim.
+      -- hence both events trigger one after another
+      command('autocmd VimSuspend  * :let g:suspend=1+get(g:, "suspend", 0)') -- increment by 1
+      command('autocmd VimResume  * :let g:suspend=get(g:, "suspend", 0)-2') -- decrement by 2
```

Comment:
use a different variable (`g:resume`) to count VimResume events. Otherwise the tests can't know if `0` means "no events fired" or "both events fired the same number of times".

Reply:
this should not resolve`suspend` to 0 as every-time nvim is resumed it will be decremented by 2. 

Therefore suspend+resume will result in `suspend` being decremented by 1 each time. 

nevertheless i think it is over-complicating things for nothing so I've made the changes like you mentioned.